src/libstd/sys/unix/pipe.rs

```diff
@@ -1,58 +1,39 @@
 use crate::io::{self, IoSlice, IoSliceMut};
 use crate::mem;
-use crate::sync::atomic::{AtomicBool, Ordering};
 use crate::sys::fd::FileDesc;
 use crate::sys::{cvt, cvt_r};
 
-use libc::c_int;
-
 ////////////////////////////////////////////////////////////////////////////////
 // Anonymous pipes
 ////////////////////////////////////////////////////////////////////////////////
 
 pub struct AnonPipe(FileDesc);
 
 pub fn anon_pipe() -> io::Result<(AnonPipe, AnonPipe)> {
-    syscall! { fn pipe2(fds: *mut c_int, flags: c_int) -> c_int }
-    static INVALID: AtomicBool = AtomicBool::new(false);
-
     let mut fds = [0; 2];
 
     // Unfortunately the only known way right now to create atomically set the
```

Comment:
It's probably not unfortunate anymore now that we can assume the existence of pipe2 :P

Reply:
Well, I guess it's still unfortunate for the few remaining `unix` targets that don't have `pipe2`. :)

But I'll reword this a little, because the condition is already covering more than just Linux.

Reply:
Actually, the change from the weak `syscall!` is probably going to fail here on the leftover targets. I think I'll change this `if cfg!` to `cfg_if!` so it's masked completely.

Reply:
That's raising a bunch of now-unused stuff... I'll clean it up more tomorrow.

Reply:
OK, I hope this is better now.